keras/metrics.py

```diff
@@ -20,6 +21,138 @@
 from .utils.generic_utils import deserialize_keras_object
+from .engine.topology import _to_snake_case
+
+
+class GlobalMetric(object):
+    """Base class for global metrics, which persist over epochs. Global Metrics
+    must inherit from this class.
+
+    # Properties:
+        __name__
+
+    # Methods:
+        update_states(self. y_true, y_pred)
+        reset_states(self)
+    """
+    def __init__(self):
+        """Initialize the Global Metric and give it a name.
+        """
+        # Instance name is class name with underscores
+        cls_name = self.__class__.__name__
+        self.__name__ = _to_snake_case(cls_name)
+
+    def __call__(self, y_true, y_pred):
+        return self.update_states(y_true, y_pred)
+
+    @abstractmethod
+    def update_states(self, y_true, y_pred):
+        """Update the state of the metric with the current values of y_true,
+        y_pred.
+
+        # Arguments
+            y_true: the true labels, either binary or categorical.
+            y_pred: the predictions, either binary or categorical.
+
+        # Returns
+            The current state of the metric.
+
+        # Raises
+            NotImplementedError: if the derived class fails to implement the method.
+        """
+        raise NotImplementedError("Method not implemented.")
+
+    @abstractmethod
+    def reset_states(self):
+        """Resets the state of the metric at the beginning of training and validation for each epoch.
+
+        # Raises
+            NotImplementedError: if the derived class fails to implement the method.
+        """
+        raise NotImplementedError("Method not implemented.")
+
+
+def reset_global_metrics(metrics):
+    """Call reset_states() for all global metrics.
+
+    # Arguments
+        metrics: a list of metric instances.
+    """
+    if metrics is not None:
+        for metric in metrics:
+            if isinstance(metric, GlobalMetric):
+                metric.reset_states()
+
+
+def get_global_metrics(metrics):
+    """ Return a list of global metrics.
+
+    # Arguments
+        metrics: a list of metric instances.
+    """
+    global_metrics = []
+    global_metric_names = []
+
+    if metrics is not None:
+        for m in metrics:
+            if isinstance(m, GlobalMetric):
+                global_metrics.append(m)
+                global_metric_names.append(serialize(m))
+
+    return global_metrics, global_metric_names
+
+
+class TruePositives(GlobalMetric):
+    """Global Metric to count the total true positives over all batches.
+
+    # Properties
+        threshold: the lower limit on y_pred that counts as a positive class prediction
+        state: the current state of the metric through the current batch
+    """
+
+    def __init__(self, threshold=None):
+        """Set the threshold and name the metric
+
+        # Keyword Arguments
+            threshold: the lower limit on y_pred that counts as a postive class prediction.
+                Defaults to 0.5
```

Comment:
If the default is 0.5, then I would suggest to put `def __init__(self, threshold=0.5):`